src/ispc/kernels/lanczos3.ispc

```diff
@@ -222,9 +220,65 @@ void clean_and_write_4_channels(varying float<4> color, varying uint64 write_add
 }
 
 /// scratch_space must be at least src_height * target_width pixels big
-export void resample_with_cache(uniform uint src_width, uniform uint src_height, uniform uint target_width, uniform uint target_height, uniform uint8 num_channels,
+export void resample_with_cache_3(uniform uint src_width, uniform uint src_height, uniform uint target_width, uniform uint target_height,
+    uniform const Cache * uniform cache, uniform uint8 scratch_space[], uniform const uint8 src_data[], uniform uint8 out_data[]) {
+    // TODO[#Kamen]: Ideally, we should split this function into two versions depending channel count, and branch only once in Rust than twice per sample.
```

Comment:
Looks like you did that now.

Anyway, is this even faster than the:

```
void resample_with_cache(uniform const uint num_channels, ...) {}

void resample_with_cache_3(...) {
   resample_with_cache(3, ...);
}

void resample_with_cache_4(...) {
   resample_with_cache(4, ...);
}
```

And instead have to duplicate _the entire body_ of `resample_with_cache()`? Implying ISPC doesn't do constant-folding? What was the perf difference?

Reply:
No, when we did the testing, the results remained the same. Duplicating the function with just changing the number of channels we read from/write to does result in a ~20% performance gain however.